src/logging.c

```diff
@@ -72,6 +103,44 @@ void rcutils_log(
   }
 }
 
+/*!
+  RCUTILS_LOGGING_ENSURE_LARGE_ENOUGH_BUFFER(
+   n, output_buffer_size, allocator, output_buffer, static_output_buffer)
+  Macro for (re)allocating a dynamic buffer if the ouput buffer doesn't have enough space for an
+  additional n characters and null terminator. Whether to allocate or re-allocate is determined by
+  if the output_buffer points to the original static_output_buffer or not. If a new buffer is
+  allocated it will be padded with NUL.
+*/
+
+#define RCUTILS_LOGGING_ENSURE_LARGE_ENOUGH_BUFFER( \
+    n, output_buffer_size, allocator, output_buffer, static_output_buffer) \
+  size_t required_output_buffer_size = strlen(output_buffer) + n + 1; \
+  if (required_output_buffer_size >= output_buffer_size) { \
```

Comment:
Why does it have to reallocate if `required_output_buffer_size` == `output_buffer_size`?

Same three lines below.

Reply:
It doesn't need to how the code is now, thanks! 136223ff4fc32e52ad7810d7441fb9fff0040a49